### ISSUE TITLE
Where method on DataFrames can't seem to take series as conditions

### ISSUE THREAD

**Issue by maximilianr** [pydata/pandas#9558]
I want to use `where` to set values across a DataFrame where a Series meets a condition.

For example:
```python
df=pd.DataFrame({'a':[1,2,3], 'b':[4,5,6]})
```
But this 
```python
df.where(df['a']==2,0)
```
returns an unedited DataFrame. 
Whereas this returns just the middle row, as you'd expect:
```python
df[df['a']==2]
```
Supplying `axis=0` or `axis=1` doesn't help. Am I doing something wrong? Or this unintended behavior?

>Versions
python: 2.7.6.final.0
pandas: 0.14.1

**Comment by TomAugspurger** [comment]
I believe the `cond` argument to `df.where` has to be the same shape as the original `df`.

**Comment by maximilianr** [comment]
@TomAugspurger, that's correct.

Is that desired behavior though? I would think that broadcasting across an index (in the same way as the final example broadcasts) would be desired.

**Comment by jreback** [comment]
@maximilianr the reason for this is that ``.where()`` is shape preserving, IOW it puts ``nan`` in the non-matching conditions.

``df[df['a']==2]`` potentially changes the shape of the result. e.g. it can drop rows.

FYI, ``df.where(df['a']==2)`` raises (an odd error message), so actually prob need to do some more checking for shape compat (right now it just does a ``.reindex()``).

**Comment by maximilianr** [comment]
@jreback Thanks!

**Comment by maximilianr** [comment]
This is closed with #10283. Closing
